### PR TITLE
[mono][jit] Enabled Ctor, Set_Item and Get_Item Vector4 intrinsics for miniJIT on ARM64

### PR DESCRIPTION
**Pull request by LeVladIonescu**
This PR contributes to https://github.com/dotnet/runtime/issues/86957.

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -2732,12 +2731,22 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
...
-		MONO_EMIT_NEW_COND_EXC (cfg, GE_UN, "ArgumentOutOfRangeException");
-		ins = emit_simd_ins (cfg, klass, OP_XINSERT_R4, dreg, args [2]->dreg);
-		ins->sreg3 = args [1]->dreg;
+		// Bounds check only if the index is out of range
+		if (index < 0 || index >= elems) {
+			MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, args [1]->dreg, len);
+			MONO_EMIT_NEW_COND_EXC (cfg, GE_UN, "ArgumentOutOfRangeException");
+		}
+
+		if (COMPILE_LLVM (cfg)) {
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:2743]: I believe that there is something that we could do to simplify this code path. Please open an issue to track it. You could say something like "Merge LLVM and non-llvm code path in emit_vector_2_3_4" in the issue.

**Review by fanyang-mono — APPROVED** [review]
LGTM. Thanks for adding the intrinsics for Vector4. :)

**Comment by fanyang-mono** [comment]
Looks like the test failures on `Build Libraries Test Run release mono linux arm64 Debug` are related. Please take a look.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -2732,12 +2731,22 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
...
+			ins = emit_simd_ins (cfg, klass, OP_XINSERT_R4, dreg, args [2]->dreg);
+			ins->sreg3 = args [1]->dreg;
+		} else {
+			ins = emit_simd_ins (cfg, klass, OP_INSERT_R4, dreg, args [2]->dreg);
+			ins->inst_c0 = index;
```
Comment (jandupej) [inline src/mono/mono/mini/simd-intrinsics.c:2748]: `set_Item` can be called with nonconstant `index`, in which case I think the mini path would not work. You may want to look at https://github.com/dotnet/runtime/blob/a0a8ce681afda6b662e450ac9532b0d4c125520c/src/mono/mono/mini/simd-intrinsics.c#L2183C11-L2183C11, where the constant case is detected (an optimized if possible). If `index` is not constant, you'll need to use `OP_XINSERT_R4`

**Review by jandupej — CHANGES_REQUESTED** [review]

**Comment by LeVladIonescu** [comment]
> Looks like the test failures on `Build Libraries Test Run release mono linux arm64 Debug` are related. Please take a look.

They were caused by not handling non constant index cases for get and set item.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 2 above)
Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c]: Agree with you. 
Updated the PR.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:2785]: https://github.com/dotnet/runtime/issues/89616

File: src/mono/mono/mini/simd-intrinsics.c (hunk 2 above)
Comment (jandupej) [reply src/mono/mono/mini/simd-intrinsics.c]: This look OK now.

**Review by jandupej — APPROVED** [review]

**Pull request merged**